coarse/api/typogrammar.py

```
@@ -0,0 +1,15 @@
+from coarse.services.typo_grammar_agent import check_typo_grammar
+from coarse.models.paperobj import PaperObj
+
+router = APIRouter()
+
+@router.post("/typogrammar")
+async def typo_grammar(paper_obj: PaperObj):
+    try:
+        result = await check_typo_grammar(paper_obj)
+        return result
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"An error occurred while processing the PDF: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)